### DISCUSSION TITLE
Instanciating the PrismaClient

### DISCUSSION THREAD

**Discussion by xarielah** [prisma/prisma#17180] [Q&A] [answered]
Hey there!

Using Nextjs (for backend I'm using next-api-decorators lib just to be clear) so I'm looking for a way to instantiate the client in an exportable class service so I can implement it in a User service or any kind of service I'd like.

I understand that any instantiation creates another instance of PrismaClient and if I dont set it to disconnect after it runs, I get my database slower and slower until it will not respond.

**Example code from NestJS that I cannot (and don't quite understand) implement in next:**
`import { INestApplication, Injectable, OnModuleInit } from '@nestjs/common';
import { PrismaClient } from '@prisma/client';

@Injectable()
export class PrismaService extends PrismaClient implements OnModuleInit {
  async onModuleInit() {
    await this.$connect();
  }

  async enableShutdownHooks(app: INestApplication) {
    this.$on('beforeExit', async () => {
      await app.close();
    });
  }
}`

**Comment by nurul3101** [comment, marked as answer]
Hello @xarielah 👋 

I haven't used the `next-api-decorators` library,  I would recommend you to instantiate PrismaClient in a Next.js App like [this](https://www.prisma.io/docs/guides/database/troubleshooting-orm/help-articles/nextjs-prisma-client-dev-practices#solution) way in a file. After creating this file, you can now import this PrismaClient instance anywhere in your Next.js pages.

Would this work for you?

**Reply by xarielah** [reply]
I actually ended up doing that this way, that is a totally sufficient solution, thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
